src/layup/utilities/file_io/ObjectDataReader.py

```diff
@@ -187,27 +198,43 @@ def _process_and_validate_input_table(self, input_table, **kwargs):
             if True then checks the data for  NaNs or nulls.
 
         """
-        pplogger = logging.getLogger(__name__)
+        logger = logging.getLogger(__name__)
 
         # Check that the table has more than one column.
         if len(input_table.dtype.names) <= 1:
             outstr = (
                 f"ERROR: While reading table {self.filename}. Only one column found. "
                 "Check that you specified the correct format."
             )
-            pplogger.error(outstr)
+            logger.error(outstr)
             sys.exit(outstr)
 
         # Check that "ObjID" is a column and is a string.
         input_table = self._validate_object_id_column(input_table)
 
-        # TODO check that format is the same across all rows?
+        # Check that the format column is present, and that all formats are the same.
+        if self._format_column_name:
+            if self._format_column_name not in input_table.dtype.names:
+                outstr = f"ERROR: While reading table {self.filename}. Format column {self._format_column_name} not found."
+                logger.error(outstr)
+                sys.exit(outstr)
+            if len(np.unique(input_table[self._format_column_name])) != 1:
+                outstr = f"ERROR: While reading table {self.filename}. Multiple formats found."
+                logger.error(outstr)
+                sys.exit(outstr)
+
+            # Check that the expected columns are present in this chunk
+            for col in REQUIRED_COLUMN_NAMES[self._format_column_name]:
+                if col not in input_table.dtype.names:
+                    outstr = f"ERROR: While reading table {self.filename}. Required column {col} not found."
+                    logger.error(outstr)
+                    sys.exit(outstr)
```

Comment:
I think REQUIRED_COLUMN_NAMES might need to be something sent in because for layup orbitfit the columns are going to be different for the CSV file/HDF file so does it make sense that required_column_names be set in the function that calls the reader fclass?

Reply:
Created https://github.com/Smithsonian/layup/issues/71 to track this work